### PR TITLE
Step 2 - Clean rolling avg input data

### PR DESCRIPTION
**Pull request by garypolzin**
# Description
Only include locations who have 2 submissions and haven't changed between care home and non res

# How to test
[Branch run](https://eu-west-2.console.aws.amazon.com/states/home?region=eu-west-2#/v2/executions/details/arn:aws:states:eu-west-2:344210435447:execution:rolling-avg-input-data-Ind-CQC-Filled-Post-Estimates-Pipeline:6b005589-e07c-4c2c-a3dc-ed63ea074e8b)

[Trello](https://trello.com/c/PwSU8I8N/876-filter-input-data-for-rolling-avg)

# Developer checklist
- [X] Unit test
- [X] Linked to Trello ticket
- [X] Documentation up to date

### CODE REVIEW

File: tests/test_file_schemas.py [hunk 1]
```diff
@@ -2523,26 +2523,31 @@ class EstimateMissingAscwdsFilledPostsSchemas:
 
 @dataclass
 class ModelPrimaryServiceRollingAverage:
+    care_home_status_count: str = "care_home_status_count"
```
Comment (MarionJHolloway) [inline tests/test_file_schemas.py:2526]: Do these want to be here or in the columns utils file?

File: tests/unit/test_model_primary_service_rolling_average.py [hunk 2]
```diff
@@ -105,8 +117,151 @@ def test_returned_column_to_average_values_match_expected(
...
+        self.returned_df = job.clean_column_to_average(self.test_df)
+        self.expected_df = self.spark.createDataFrame(
+            Data.expected_clean_column_to_average_rows,
+            Schemas.clean_column_to_average_schema,
+        )
+
+    def test_clean_column_to_average_returns_original_columns(self):
+        self.assertEqual(self.returned_df.columns, self.test_df.columns)
+
+    def test_clean_column_to_average_is_not_nulled_when_submitted_submitted_more_than_once_and_consistent_care_home_status(
```
Comment (MarionJHolloway) [inline tests/unit/test_model_primary_service_rolling_average.py:143]: Submitted is repeated in this name

File: tests/unit/test_model_primary_service_rolling_average.py [hunk 3]
```diff
@@ -105,8 +117,151 @@ def test_returned_column_to_average_values_match_expected(
...
+        self.returned_data = self.returned_df.sort(IndCqc.location_id).collect()
+        self.expected_data = self.expected_df.collect()
+
+    def test_calculate_submission_count_returns_expected_columns(self):
+        self.assertEqual(
+            sorted(self.returned_df.columns),
+            sorted(self.expected_df.columns),
+        )
+
+    def test_returned_submission_values_match_expected(
```
Comment (MarionJHolloway) [inline tests/unit/test_model_primary_service_rolling_average.py:236]: add "when_location_does_not_have_multiple_care_home_statuses"

**Review by MarionJHolloway — COMMENTED** [review]
Looks good, a few small comments

File: tests/test_file_schemas.py (hunk 1 above)
Comment (garypolzin) [reply tests/test_file_schemas.py:2526]: I move these in a later PR so will leave this for now

**Review by MarionJHolloway — APPROVED** [review]

**Pull request merged**